src/api/schemas.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2022 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Testflinger v1 OpenAPI schemas
+"""
+
+from apiflask import Schema, fields
+from apiflask.validators import OneOf
+
+
+class AgentIn(Schema):
+    """Agent data input schema"""
+
+    state = fields.String(required=False)
+    queues = fields.List(fields.String(), required=False)
+    location = fields.String(required=False)
+    job_id = fields.String(required=False)
+    log = fields.List(fields.String(), required=False)
+
+
+class ActionIn(Schema):
+    """Action data input schema"""
+
+    action = fields.String(required=True, validate=OneOf(["cancel"]))
+
+
+class Job(Schema):
+    """Job schema"""
+
+    job_id = fields.String(required=False)
+    parent_job_id = fields.String(required=False)
+    name = fields.String(required=False)
+    job_queue = fields.String(required=True)
+    global_timeout = fields.Integer(required=False)
+    output_timeout = fields.Integer(required=False)
+    allocation_timeout = fields.Integer(required=False)
+    provision_data = fields.Dict(required=False)
+    test_data = fields.Dict(required=False)
+    allocate_data = fields.Dict(required=False)
+    recover_data = fields.Dict(required=False)
```

Comment:
they are variable - some are dependent on the type of device. That's something to think about for later down the road with v2 api I think. For example, provisioning a device with muxpi you need to specify a url key. For maas, you need to specify a distro key, because it must already be imported and can't use a generic url.